prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -178,48 +180,108 @@ static int handle_poll_list(struct poll_fd_mgr *poll_mgr)
 	return ret;
 }
 
-static void handle_connect(struct poll_fd_mgr *poll_mgr,
-			   struct poll_fd_info *poll_info)
+static int handle_connect(struct poll_fd_mgr *poll_mgr,
+			  int index)
 {
 	struct tcpx_ep *ep;
-	struct fi_eq_cm_entry cm_entry;
+	struct poll_fd_info *poll_info = &poll_mgr->poll_info[index];
+	struct fi_eq_cm_entry *cm_entry;
+	struct tcpx_conn_req conn_req, conn_resp;
 	struct fi_eq_err_entry err_entry;
 	socklen_t len;
+	uint16_t cm_data_sz;
 	int ret, status;
 
 	assert(poll_info->fid->fclass == FI_CLASS_EP);
 	ep = container_of(poll_info->fid, struct tcpx_ep, util_ep.ep_fid.fid);
 
-	len = sizeof(status);
-	ret = getsockopt(ep->conn_fd, SOL_SOCKET, SO_ERROR, &status, &len);
-	if (ret < 0 || status) {
-		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL, "connection failure\n");
+	if (ESTABLISH_CONN == poll_info->state) {
```

Comment:
nit: Here and elsewhere, please use forward logic.  I.e. we're checking that state == ESTABLISH_CONN, not the reverse.  This improves readability.